RELEASENOTES.md

```diff
@@ -1,18 +1,45 @@
+- Augmented `submit` method to include additional details on the status of the command. [`79e9085dd`](https://github.com/ripple/rippled/commit/79e9085dd1eb72864afe841225b78ec96e72b5ca)
+- Improved `tx` method response with additional details on ledgers searched. [`47501b7f9`](https://github.com/ripple/rippled/commit/47501b7f99d4103d9ad405e399169fc251161548)
+- New `validator_info` method which returns information pertaining to the current validator's keys, manifest sequence, and domain. [`3578acaf0`](https://github.com/ripple/rippled/commit/3578acaf0b5f2d27ddc33f5b4cc81d21be1903ae)
+- New `manifest` method which looks up manifest information for the specified key (either master or ephemeral). [`3578acaf0`](https://github.com/ripple/rippled/commit/3578acaf0b5f2d27ddc33f5b4cc81d21be1903ae)
+- Introduce handshake protocol for compression negotiation (compression is not implemented at this point) and other minor improvements. [`f6916bfd4`](https://github.com/ripple/rippled/commit/f6916bfd429ce654e017ae9686cb023d9e05408b)
+- Remove various old conditionals introduced by amendments. [`(51ed7db00`](https://github.com/ripple/rippled/commit/51ed7db0027ba822739bd9de6f2613f97c1b227b), [`6e4945c56)`](https://github.com/ripple/rippled/commit/6e4945c56b1a1c063b32921d7750607587ec3063)
+- Add `getRippledInfo` info gathering script to `rippled` Linux packages. [`acf4b7889`](https://github.com/ripple/rippled/commit/acf4b78892074303cb1fa22b778da5e7e7eddeda)
+
+**Bug Fixes**
+- The `fixQualityUpperBound` amendment which fixes a bug in unused code for estimating the ratio of input to output of individual steps in cross-currency payments. [`9d3626fec`](https://github.com/ripple/rippled/commit/9d3626fec5b610100f401dc0d25b9ec8e4a9a362)
+- The `RequireFullyCanonicalSig` amendment which changes the signature requirements for the XRP Ledger protocol so that non-fully-canonical signatures are no longer valid. This protects against transaction malleability on all transactions, instead of just transactions with the tfFullyCanonicalSig flag enabled. Without this amendment, a transaction is malleable if it uses a secp256k1 signature and does not have tfFullyCanonicalSig enabled. Most signing utilities enable tfFullyCanonicalSig by default, but there are exceptions. With this amendment, no single-signed transactions are malleable. (Multi-signed transactions may still be malleable if signers provide more signatures than are necessary.) All transactions must use the fully canonical form of the signature, regardless of the tfFullyCanonicalSig flag. Signing utilities that do not create fully canonical signatures are not supported. All of Ripple's signing utilities have been providing fully-canonical signatures exclusively since at least 2014. For more information. [`ec137044a`](https://github.com/ripple/rippled/commit/ec137044a014530263cd3309d81643a5a3c1fdab)
```

Comment:
I asked the same question over in https://github.com/ripple/dev-blog/pull/40#discussion_r400512220:

Is this really a bug being fixed? Am I wrong that allowing non-canonical sigs was an early intentional design decision?

Reply:
_(Copying my comment over here, where the public can see it.)_

Allowing non-fully-canonical signatures was not an intentional design decision, to my understanding. There have definitely been significant advances in the state of elliptic curve cryptography since 2009 (when Bitcoin's signing code, which the XRP ledger copied, was written) and 2011 (when @JoelKatz started writing the XRPL code). For example, deterministic nonces¹, which have since become an important security precaution for ECDSA signatures, weren't specified [until 2013](https://tools.ietf.org/html/rfc6979).

I believe the way it happened was that only after the signature code had been put in place did it become apparent that the rules for "canonical" signatures were not sufficient to make sure any given signature could not be malleable. When this became known, the devs added the tfFullyCanonical flag to provide protection, and had the official signing code turn this flag on by default whenever possible. I don't know if the idea at the time was for this to be a transitional step to provide time to upgrade any code that made non-fully-canonical signatures; but I think it served that purpose. At this point, the benefits of maintaining that legacy support are outweighed by the benefits of protecting more transactions against malleability.

Ultimately I would suggest categorizing this as an "improvement" that is neither a feature nor a bug fix.

¹ In contrast to fully-canonical-signatures, which you can recognize with only the signature and the public key necessary to confirm it, I think there is no way to know whether a signature uses a deterministic nonce unless you know the private key. So, unfortunately, the ledger cannot enforce deterministic nonces in the way it enforces fully-canonical signatures.

Reply:
Updated and moved it to the section above.